aten/src/ATen/native/BlasKernel.cpp

```diff
@@ -260,6 +260,22 @@ static inline double reduce(float16x8_t x[F16_REGISTERS_PER_ITERATION]) {
   for (int i = 0; i < offset; ++i) {
     x[i] = vaddq_f16(x[i], x[offset + i]);
   }
+  offset /= 2;
```

Comment:
Wouldn't nested loop (or c10::LoopUnroll) would be better here?

Reply:
what is c10::LoopUnroll?

and yes I think this could be cleaned up, happy to do that but I'd like to take the opportunity to do a specific cleanup pass

Reply:
> cleaned up

https://github.com/pytorch/pytorch/pull/127016